### ISSUE TITLE
2.2.0 error while exporting tiles

### ISSUE THREAD

**Issue by skemaikin** [Geodan/pg2b3dm#140]
Hi!
I got an error while exporting tiles.

```
docker run --rm -v $(pwd)/output:/app/output -it geodan/pg2b3dm:2.2.0 -h 192.168.0.250 -p 5432 -U postgres -d postgres -t relief -c geom_triangle_4326 -a 'ogc_fid' --shaderscolumn shaders --create_gltf false --use_implicit_tiling false
```
Output:
```
Tool: pg2b3dm 2.2.0.0
Options: -h 192.168.0.250 -p 5432 -U postgres -d postgres -t relief -c geom_triangle_4326 -a ogc_fid --shaderscolumn shaders --create_gltf false --use_implicit_tiling false
Password for user postgres: 
Start processing 2024-01-26T11:02:21....
Input table: relief
Input geometry column: geom_triangle_4326
Spatial reference of relief.geom_triangle_4326: 4326
Spatial index detected on relief.geom_triangle_4326
Query bounding box of relief.geom_triangle_4326...
Bounding box for relief.geom_triangle_4326 (in WGS84): 30.21374078, 59.82277929, 30.51813451, 60.03005767
Height values: [-0 m - 11.5 m]
Default color: #FFFFFF
Default metallic roughness: #008000
Doublesided: True
Create glTF tiles: False
Attribute columns: ogc_fid
Center (wgs84): 30.365937648867476, 59.92641847814146
Starting Cesium mode...
Translation ECEF: 2764631,1619790.875,5496373.5
3D Tiles version: 1.0
Lod column: 
Geometric errors: 2000,0
Refinement: REPLACE
Geometric errors used: 2000,0
Add outlines: False
Use 3D Tiles 1.1 implicit tiling: False
Maximum features per tile: 1000
Start generating tiles...
Creating tile: 6_3_14.b3dm  Unhandled exception. System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at Wkb2Gltf.GlbCreator.GetGlb(List`1 triangles, String copyright, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean defaultDoubleSided, Dictionary`2 attributes, Boolean createGltf, Boolean doubleSided) in /src/wkb2gltf.core/GlbCreator.cs:line 77
   at Wkb2Gltf.TileCreator.GetTile(Dictionary`2 attributes, List`1 triangleCollection, String copyright, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/wkb2gltf.core/TileCreator.cs:line 12
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 102
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 84
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 84
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 84
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 84
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 84
   at pg2b3dm.QuadtreeTiler.GenerateTiles(BoundingBox bbox, Tile tile, List`1 tiles, Int32 lod, Boolean addOutlines, String defaultColor, String defaultMetallicRoughness, Boolean doubleSided, Boolean createGltf) in /src/b3dm.tileset/QuadtreeTiler.cs:line 84
   at pg2b3dm.Program.<>c__DisplayClass3_0.<Main>b__0(Options o) in /src/pg2b3dm/Program.cs:line 187
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at pg2b3dm.Program.Main(String[] args) in /src/pg2b3dm/Program.cs:line 28
```

It's very strange because all my geometries are in 4326 and options includes ```--create_gltf false --use_implicit_tiling false```.

**Comment by bertt** [comment]
hmm never seen this, does it happen at the first tile? what if you omit parameters like  -a ogc_fid and/or --shaderscolumn shaders ?

**Comment by skemaikin** [comment]
According to the debugging results:

1. there is an attempt to create a tile that receives a single triangle;  
2. this triangle turns out to be degenerate and `AddTriangleWithBatchId/AddTriangleWithFeatureId`-method cannot add this triangle to the collection;
3. as a result, the `scene` object becomes incorrect and the code: `model.LogicalNodes.First().LocalTransform = new SharpGLTF.Transforms.AffineTransform(localTransform);` (wkb2gltf.core/GlbCreator.cs:64) throws an exception.

It is necessary to check the cases where `model.LogicalNodes.Count` is 0.

**Comment by bertt** [comment]
Ah yes I’ve changed the degenerated triangles. I’ll add some error handling

**Comment by bertt** [comment]
fix in 2.2.1 https://github.com/Geodan/pg2b3dm/pull/141

tested with table:

CREATE TABLE issue140_4326 (geom geometry)
insert into issue140_4326(geom) values (st_setsrid(st_geomfromtext('POLYGON((0 0 0,0 0 0,0 0 0,0 0 0))'),4326));

**Comment by skemaikin** [comment]
Now it works fine!!!
